lib/service/signals.go

```diff
@@ -323,6 +331,15 @@ func (process *TeleportProcess) createListener(typ ListenerType, address string)
 		}
 		return nil, trace.Wrap(err)
 	}
+
+	// The default behavior for unix listeners is to delete the file when the
+	// listener closes (unlinking). However, if the process forks, the file
+	// descriptor will be gone when its parent process exists, causing the new
+	// listener to have no socket file.
+	if unixListener, ok := listener.(*net.UnixListener); ok {
+		unixListener.SetUnlinkOnClose(false)
+	}
```

Comment:
It's much simpler to always leave the socket on disk if it can't lead to a proliferation of socket files IMO. There's no guarantee that the child will correctly finish initialization, which means that we need to handle the situation where we need to "gain back" the control over the unlinking of the socket etc.

Reply:
It's much simpler to always leave the socket on disk if it can't lead to a proliferation of socket files IMO. There's no guarantee that the child will correctly finish initialization, which means that we need to handle the situation where we need to "gain back" the control over the unlinking of the socket etc.

Reply:
Agreed that cleaning up before starting the listener is required either way. I guess I just have to change my expectation that if the socket file exists in the dir does not necessarily mean that the process is running. Well, the process should be mostly running anyway. Simpler wins.